services/provider/server/server.go

```diff
@@ -412,10 +418,22 @@ func validateTicket(ticket string, pubKey *rsa.PublicKey) error {
 	if err != nil {
 		return fmt.Errorf("failed to decode payload. %v", err)
 	}
+
+	var ticketDate TicketDate
+	err = json.Unmarshal(message, &ticketDate)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal message. %v", err)
+	}
+
 	signature, err := base64.StdEncoding.DecodeString(ticketArr[1])
 	if err != nil {
-		return fmt.Errorf("failed to decode signature. %v", err)
+		return fmt.Errorf("failed to decode onboarding ticket %s signature: %v", ticketDate.ID, err)
+	}
+
+	if ticketDate.ExpirationDate < time.Now().Unix() {
+		return fmt.Errorf(" Onboarding ticket %s is expired ", ticketDate.ID)
 	}
+	klog.Infof("onboarding ticket %s has been verified successfully", ticketDate.ID)
```

Comment:
This need to be the last thing on the function, and there is another variation of it already there `Successfully verified onboarding ticket`, but with no ID